source/mir/random/generic/package.d

```diff
@@ -0,0 +1,261 @@
+/**
+License: $(LINK2 http://boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Authors: Sebastian Wilzbach, Ilya Yaroshenko
+
+The Transformed Density Rejection with Inflection Points (Tinflex) algorithm
+can sample from arbitrary distributions given its density function f, its
+first two derivatives and a partitioning into intervals with at most one inflection
+point.
+
+These can be easily found by plotting `f''`.
+$(B Inflection point) can be identified by observing at which points `f''` is 0.
+
+$(UL
+    $(LI $(F_TILDE) is entirely concave (`f''` is entirely negative))
+    $(LI $(F_TILDE) is entirely convex (`f''` is entirely positive))
+    $(LI $(F_TILDE) contains one inflection point (`f''` intersects the x-axis once)
+)
+
+In exact terms the algorithm requires:
+
+$(UL
+    $(LI Continuous density function $(F_TILDE).)
+    $(LI Continuous differentiability of $(F_TILDE) except in a finite number of
+      points which need to have a one-sided derivative.)
+    $(LI $(B Doubled) Continuous differentiability of $(F_TILDE) except in a finite number of
+      points which need to be inflection points.)
+    $(LI At most one inflection point per interval)
+)
+
+References:
+    Transformed Density Rejection with Inflection Points
+
+    Botts, Carsten, Wolfgang Hörmann, and Josef Leydold.
+    "$(LINK2 http://epub.wu-wien.ac.at/3158/1/techreport-110.pdf,
+    Transformed density rejection with inflection points.)"
+    Statistics and Computing 23.2 (2013): 251-260.
+
+Macros:
+    F_TILDE=g(x)
+*/
+module mir.random.generic;
+
+import mir.random.generic.types : IntervalPoint;
+
+import std.traits : ReturnType;
+import std.random : isUniformRNG;
+
+
+/**
+The Transformed Density Rejection with Inflection Points (Tinflex) algorithm
+can sample from arbitrary distributions given its density function f, its
+first two derivatives and a partitioning into intervals with at most one inflection
+point.
+
+Params:
+    f0 = probability density function of the distribution
+    f1 = first derivative of f0
+    f1 = second derivative of f0
+    c = T_c family
+    points = non-overlapping partitioning with at most one inflection point per interval
+    rho = efficiency of the Tinflex algorithm
+
+Returns:
+    Tinflex Generator.
+*/
+Tinflex!(F0, S) tinflex(F0, F1, F2, S)
+               (in F0 f0, in F1 f1, in F2 f2,
```

Comment:
hmm, ok, lets use `in` for callable function because `opCall` methods are `cosnt`